### DISCUSSION TITLE
uiautomator2 can't find waitActivity because it closes too quickly

### DISCUSSION THREAD

**Discussion by dedevXamarin** [appium/.github#3] [Q&A] [answered]
I'm running appium server v2.2.1 with Visual Studio Appium WebDriver v4.3.2. When my app launches on the device, Appium Server tells me that it's looking for the Main Activity, but finds the LoginActivity. It appears that Main Activity is closing too quickly - Is there any way to force appium to look for the Login Activity rather than whatever the main activity is?

**Comment by mykola-mokhnach** [comment]
Yes, this is possible by providing appWaitActivity. You can even disable waiting completely. Read https://github.com/appium/appium-uiautomator2-driver/blob/master/docs/activity-startup.md for more details.

**Reply by dedevXamarin** [reply]
This just doesn't seem to work. I added ("appWaitDuration", "false") to my capabilities, but it still waits. When I add AppWaitActivity, it waits for something OTHER than the activity I specify. These two capabilities seem to have no impact on waiting, or waiting for a specific activity. Is this a problem with the Appium .Net driver?

**Reply by mykola-mokhnach** [reply, marked as answer]
appWaitDuration must be a number, not a boolean.
Also, in order to skip waiting completely the appWaitForLaunch capability must be set to false

**Reply by dedevXamarin** [reply]
Thank you!!! you fixed my issue :) 

I set appWaitForLaunch to false.
